Week-1/Exercise1.1/readme.md

```diff
@@ -0,0 +1,263 @@
+
+In practice, websites fetch a lot of additional data (via additional requests and responses) which are kicked off once the first response arrived.
+
+How does that work?
+
+Well, the HTML code of the first response simply contains instructions to fetch more data via new requests - and the browser understands these instructions:
+
+```
+<link crossorigin="anonymous" media="all" rel="stylesheet" href="https://github.githubassets.com/assets/dashboard-74d536a2af1a.css" />
+```
+
+Together with CSS, the browser is able to display webpages like this:
+
+<br>
+
+![alt text](fullpagewithcss.PNG)
+
+<br>
+
+There actually is another programming language involved (this time, it really is a programming language!): JavaScript.
+
+It's not always visible but all dynamic content you find on a website (e.g. tabs, overlays etc.) is actually only possible because of JavaScript. It allows web developers to define code that runs in the browser (not on the server), hence JavaScript can be used to change the website whilst the user is viewing it.
+
+These are the four steps that are always involved when you enter a page address like github.com and you thereafter see the website content in your browser.
+
+#### Server-side vs Browser-side
+
+From the four steps above, you learned that we can differentiate two core "sides" when talking about the web: Server-side and Browser-side (or: Client-side since we can also access the internet without a browser - see below!).
+
+If you're interested in becoming a web developer, it's important to know that you use different technologies and programming languages for these sides.
+
+#### Server-side
+
+You need server-side programming languages - i.e. languages that don't work in the browser but that can run on a normal computer (a server is in the end just a normal computer).
+
+Examples would be:
+
+Node.js
+PHP
+Python
+
+<br>
+<hr>
+
+#### Important:
+
+With the exception of PHP, you can also use these programming languages for other purposes than web development.
+
+Whilst Node.js is indeed primarily used for server-side programming (though it's technically not limited to that), Python is also very popular for data science and machine learning.
+
+<hr>
+<br>
+
+#### Browser-side
+
+In the browser, there are exactly three languages/ technologies you need to learn. But whilst the server-side languages were alternatives, these three browser-side languages are all mandatory to know and understand:
+
+HTML (for the structure)
+
+CSS (for the styling)
+
+JavaScript (for dynamic content)
+
+#### "Behind the Scenes" Internet
+
+Thus far, we discussed websites. I.e. the case where you enter a URL (e.g. "github.com/jasham") into the browser and you get a website in return.
+
+But the internet is more than that. Indeed, you use it for more than just that every day!
+
+The core idea of requests and responses is always the same. But not every response is necessarily a website. And not every request wants a website.
+
```

Comment:
@jasham This looks good to me, but its mostly covering the details involved in what is html, css and JS and the technologies which is used to build those.

I would rather want to cover on the steps involved behind the scenes once a html is received by the browser. What all steps are involved to show that webpage to the user.

Reply:
@RajeshSivanesan I will add those details as well.